webapp/src/components/AssetProvider/AssetProvider.tsx

```diff
@@ -30,7 +33,15 @@ const AssetProvider = (props: Props) => {
           throw new Error(`Invalid Asset type ${type}`)
       }
     }
-  }, [contractAddress, tokenId, type, onFetchNFT, onFetchItem, rentalStatus])
+  }, [
+    contractAddress,
+    tokenId,
+    type,
+    onFetchNFT,
+    onFetchItem,
+    rentalStatus,
+    isLoadingFeatureFlags
+  ])
 
   return <>{children(asset, order, rental, isLoading)}</>
```

Comment:
We might want to add to the `isLoading` flag that is passed down, the loading of the feature flags:
```suggestion
  return <>{children(asset, order, rental, isLoading || (isLoadingFeatureFlags && type === AssetType.NFT))}</>
```

Reply:
good idea! added